### PR TITLE
add weekly measures

### PR DESCRIPTION
**Pull request by AbodunrinAminu-Manchester**
This pull request is meant to look at weekly virtual consultation in three years (2019, 2020 and 2021) starting from 1st of April of each year. We have adjusted the denominator to be has_appointment because we are interested in the ratio of virtual of the appointment instead of registration. 
1. Change in denominator
2. three virtual consultation measures (2019, 2020 and 2021)
3. three generate dataset curation (2019, 2020 and 2021)

### CODE REVIEW

File: analysis/measures_definition.py [hunk 1]
```diff
@@ -73,13 +73,16 @@
 
 # Demographic variables and other patient characteristics
 # Define variable that checks if a patients is registered at the start date
-has_registration = practice_registrations.for_patient_on(
```
Comment (milanwiedemann) [inline analysis/measures_definition.py:76]: `has_registration` is common variable name and I think we should stick with it unless there's a good reason not to.

File: analysis/measures_definition.py [hunk 2]
```diff
@@ -73,13 +73,16 @@
...
+registration = practice_registrations.for_patient_on(
     INTERVAL.start_date
 ).exists_for_patient()
 
 age = patients.age_on(INTERVAL.start_date)
 age_greater_equal_65 = age >= 65
 
+# Define population to be registered and above 18 years old
+has_registration = registration & (age > 18)
```
Comment (milanwiedemann) [inline analysis/measures_definition.py:84]: `has_registration` here is not just registration but also age. was there a reason you changed the names here?

File: analysis/measures_definition.py [hunk 3]
```diff
@@ -117,71 +120,64 @@
...
-
-measures.define_measure(
-    name="appointments_during_monthly",
-    numerator=has_appointment,
-    denominator=denominator,
-    group_by={"age_greater_equal_65": age_greater_equal_65},
-    intervals=months(6).starting_on("2020-04-01"),
-)
+# Define population denominator
+denominator = has_appointment
```
Comment (milanwiedemann) [inline analysis/measures_definition.py:124]: now you're no longer including the age restriction and registration in your population, I'm not sure this is what you want to do. you probably want:

denominator = has_registration & (age > 18) & has_appointment

**Review by milanwiedemann — CHANGES_REQUESTED** [review]
Changing to weekly intervals looks good. I think the new variable names are not good descriptions of what they represent and I'm also not sure if the change to your denominator is what you want.

File: analysis/measures_definition.py (hunk 3 above)
Comment (AbodunrinAminu-Manchester) [reply analysis/measures_definition.py:124]: This very helpful!

Comment (AbodunrinAminu-Manchester) [reply analysis/measures_definition.py:124]: I am not sure what naming convention will be appropriate. I have changed the denominator accordingly.

**Pull request merged**